arboretum/kubernetes/README.md

```diff
@@ -16,7 +16,80 @@ to include the fetchers and checks from this library in your downstream project.
 
 ## Fetchers
 
-Fetchers coming soon...
+### Cluster Resource
+
+* Class: [ClusterResourceFetcher][fetch-cluster-resource]
+* Purpose: Write the resources of **stand-alone** Kubernetes clusters to the
+evidence locker.  **NOTE:** Do not use this fetcher for managed clusters.
+Instead use the [IBM Cloud cluster list fetcher][ibm-cloud-cluster-list-fetcher].
+* Behavior: Retrieve stand-alone Kubernetes cluster resource data for the provided
+list of clusters.  TTL is set to 1 day.
+* Configuration elements:
+  * `org.kubernetes.cluster_resources.clusters`
+    * Required
+    * List of dictionaries:
+      * `label`: An arbitrary label to specify token in credentials file
+      * `server` A URL as a string to a Kubernetes stand-alone cluster
+  * `org.kubernetes.cluster_resources.types`
+    * Optional
+    * List of resource types as strings
+      * NOTE: For core group API resources, the resource name must be in
+      _plural form_.  See example below.
+      * NOTE: For other "custom" API resources, the resource name must be in the
+      following format: `"APIGROUP/VERSION/NAME"`.  See example below.
+    * Use if looking to override the default of `["nodes", "pods", "configmaps"]`.
+    Otherwise do not include.
+* Expected configuration:
+
+  ```json
+  {
+    "org": {
+      "kubernetes": {
+        "cluster_resources": {
+          "clusters": [
+            {
+              "label": "devcluster1",
+              "server": "https://myserver1:30000"
+            }
+          ],
+          "types": ["nodes", "pods", "apigroup.example.com/v1/mycustom"]
+        }
+      }
+    }
+  }
+   ```
+
+* Required credentials:
+  * A token of a kubernetes service account is required. Following example
+    shows how to create a token bound with a service account.
+
+    ```sh
+    $ kubectl create serviceaccount auditree_account # this creates both serviceaccount and its secret
+    $ kubectl get serviceaccount auditree_account -ojsonpath='{.secrets[0].name}' | xargs kubectl get secret -ojsonpath='{.data.token}'
+    ZXlKaG... # use this string as a token
+    ```
```

Comment:
Is auditree_account an example of some sort and how does it correlate with the cluster label from your configuration?  The `#`ed comments don't really provide any help at the moment.  It also seems like the example token you're using here is the same token you reference in the credentials file which I believe maps to the cluster label in your configuration?  If so then it should be clear how we got here from the config and how we get to the credentials file.

The docs need to walk the reader through setting up the configuration of this fetcher, how it ties to this token creation and then how that token is entered into the credentials file.  The best way to do that is to provide a cohesive story from config write up/example to token generation write up/example to credentials file write up/example.

Reply:
I updated here as follows; is this clearer than the previous one?

https://github.com/tmishina/auditree-arboretum/blame/9-cluster-resource-fetcher/arboretum/kubernetes/README.md#L62-L93